recipes/psipred/perlwrapper.patch

```diff
@@ -0,0 +1,179 @@
++		exit $status;
++	}
++	
++	return [File::Spec->rel2abs($filename.'.ss2'), File::Spec->rel2abs($filename.'.horiz')]
++}
++
++sub blast {
++	print "Running PSI-BLAST with sequence $inputfile ...\n";
++		
++	my $blastdatadir = "";
++	my $cmd = $ncbidir.'blastpgp -b 0 -v 5000 -j 3 -h 0.001 -d '.$dbname.' -i '.$tmproot.'/input -C '.$tmproot.'/psiblast.chk &> '.$tmproot.'/psiblast.blast';
++	qx($cmd);
++	my $status = $?;
++	if ($status != 0) {
++		system("tail ${tmproot}/psiblast.blast");
++		print "FATAL: Error whilst running blastpgp - script terminated!\n";
++		exit $status
++	}
++
++	print "Predicting secondary structure...\n";
++	qx(echo 'psiblast.chk' > $tmproot/psiblast.pn);
++	qx(echo 'input' > $tmproot/psiblast.sn);
++	
++	$cmd = $ncbidir.'makemat -P '.$tmproot.'/psiblast';
++	qx($cmd);
++	$status = $?;
++	if ($status != 0) {
++		print "FATAL: Error whilst running makemat - script terminated!\n";
++		exit $status;
++	}
++}
++
++sub get_arguments {
++	my $result = GetOptions (
++				"d=s" => \$dbname,
++				"n=s" => \$ncbidir,
++				"mtx=i" => \$mtx,
++		        "h"  => sub {&usage;});
++
++	&usage if (!$ARGV[0]);
++	($inputfile) = @ARGV;
++	
++	# add a / to the end of dir if it's not empty or already existing
++	foreach my $dir ($ncbidir, $execdir, $datadir) {
++		$dir .= "/" if (length($dir) > 0) && ($dir !~ m|/$|);
++	}
++	
++	# check presents of blastpgp
++	my $test = qx(${ncbidir}blastpgp -version 2>&1);
++	if ((not defined $test) || ($test !~ m/Number of database sequences to show one-line descri/)) {
++		print STDERR "Can't find the program blastpgp in the NCBI directory $ncbidir\n"; 
++		print STDERR "Please pass the correct NCBI location using the -n parameter or modify\n";
++		print STDERR "the value at the top of the script..\n\n";
++		exit 1;
++	}
++	
++	# check presents makemat
++	$test = qx(${ncbidir}makemat 2>&1);
++	if ((not defined $test) || ($test !~ m/FATAL ERROR: Unable to open profiles file stdin\.pn/)) {
++		print STDERR "Can't find the program makemat in the NCBI directory $ncbidir\n"; 
++		print STDERR "Please pass the correct NCBI location using the -n parameter or modify\n";
++		print STDERR "the value at the top of the script.\n\n";
++		exit 1;	
++	}
++	
++	# check if blast database is present
++	if ($mtx != 1) {
++		# guess blast data dir
++		if (not exists $ENV{'BLASTMAT'}) {
++			my ($dir) = (qx(which blastpgp) =~ m|^(.*?)bin/blastpgp$|);
++			$ENV{'BLASTMAT'} = $dir.'/data/';
++		}
++		
++		if ((not defined $dbname) or ($dbname eq '')) {
++			print STDERR "Please provide a PSI-BLAST database name via -d!\n";
++			exit 1;
++		}
++		if (qx(ls ${dbname}* 2>&1) =~ m/No such file or directory/) {
++			print STDERR "The database name for PSI-BLAST searches has not been set correctly.\n"; 
++			print STDERR "Please pass it using the -d parameter or modify the value at the top of the script.\n\n";
++			exit 1;
++		}
++	}
++	
++	# check if input file exist
++	if (! -e $inputfile) {
```

Comment:
No problem! Your Perl style is not necessarily the slimmest, but it is easier to understand and maintain. That's also why Perl is replaced by Python.